src/__tests__/labs/ViewLabs.test.tsx

```diff
@@ -162,36 +167,65 @@ describe('View Labs', () => {
     })
   })
 
-  describe('sort Request', () => {
-    let findAllSpy: any
-    beforeEach(async () => {
-      const store = mockStore({
-        title: '',
-        user: { permissions: [Permissions.ViewLabs, Permissions.RequestLab] },
+  describe('search functionality', () => {
+    beforeEach(() => jest.useFakeTimers())
+
+    afterEach(() => jest.useRealTimers())
+
+    it('should search for labs after the search text has not changed for 500 milliseconds', () => {
+      const searchLabsSpy = jest.spyOn(labsSlice, 'searchLabs')
+      let wrapper: ReactWrapper
+      let history: any
+      const expectedLab = {
+        id: '1234',
+        type: 'lab type',
+        patientId: 'patientId',
+        status: 'requested',
+        requestedOn: '2020-03-30T04:43:20.102Z',
+      } as Lab
+
+      beforeEach(async () => {
+        const store = mockStore({
+          title: '',
+          user: { permissions: [Permissions.ViewLabs, Permissions.RequestLab] },
+          labs: { labs: [expectedLab] },
+        })
+        history = createMemoryHistory()
+
+        jest.spyOn(LabRepository, 'findAll').mockResolvedValue([expectedLab])
+        await act(async () => {
+          wrapper = await mount(
+            <Provider store={store}>
+              <Router history={history}>
+                <ViewLabs />
+              </Router>
+            </Provider>,
+          )
+        })
+
+        searchLabsSpy.mockClear()
+        const expectedSearchText = 'search text'
+
+        act(() => {
+          ;(wrapper.find(TextInput).prop('onChange') as any)({
+            target: {
+              value: expectedSearchText,
+            },
+            preventDefault(): void {
+              // noop
+            },
```

Comment:
I prefer to doing this in two lines so that the weird semicolon doesn't show. 

Also, I think it makes for cleaner code to use `jest.fn()` rather than a noop function

```ts
const onClick = wrapper.find(TextInput).prop('onChange') as any
onClick({
 target: {
    value: expectedSearchText
  },
  preventDefault: jest.fn()
})
```